app/controllers/concerns/can_renew_if_possible.rb

```diff
@@ -4,18 +4,29 @@ module CanRenewIfPossible
   extend ActiveSupport::Concern
 
   included do
-    after_action :renew_if_possible, only: :create
-
     private
 
     def renew_if_possible
-      return unless @resource.is_a?(WasteCarriersEngine::RenewingRegistration)
+      return false unless can_renew?
 
-      renewal_completion_service = WasteCarriersEngine::RenewalCompletionService.new(@resource)
       renewal_completion_service.complete_renewal
+
+      true
     rescue StandardError => e
       Airbrake.notify(e, reg_identifier: @resource.reg_identifier)
       Rails.logger.error e
+
+      false
```

Comment:
Codeclimate is complaining about this path not being test-covered, but I am not going to add tests here as it would be a lot of code for very little gains.

Reply:
I can live with that. This rescue is part of efforts to understand why things get `STUCK` and so may come out in the future anyway.